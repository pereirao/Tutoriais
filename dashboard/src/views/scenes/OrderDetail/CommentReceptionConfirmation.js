// @flow

import React from 'react';
import {connect} from 'react-redux';
import I18n from '../../../localization';
import {comment_actions} from '../../../business/comment';
import style_sheet from './OrderDetail.module.css';

type CommentReceptionConfirmationProps = { order_id: string, addComment: (string, Object) => void };
export const CommentReceptionConfirmation = ({order_id, addComment}: CommentReceptionConfirmationProps) => {
  const submitUnderstanding = (does_understand: boolean) => (
    addComment(order_id, I18n.t(`ui.body.comment_understanding.${does_understand}`))
  );

  return (
    <div className={style_sheet.commentReceptionConfirmation}>
      <div className={style_sheet.receptionConfirmationButton} onClick={() => submitUnderstanding(false)}>
        {I18n.t('ui.button.negative_understanding')}
      </div>
      <div className={style_sheet.receptionConfirmationButton} onClick={() => submitUnderstanding(true)}>
        {I18n.t('ui.button.positive_understanding')}
      </div>
    </div>
  );
};

const CommentReceptionConfirmationDTP = {addComment: comment_actions.addComment};
const CommentReceptionConfirmationContainer = connect(null, CommentReceptionConfirmationDTP)(CommentReceptionConfirmation);

export default CommentReceptionConfirmationContainer;
