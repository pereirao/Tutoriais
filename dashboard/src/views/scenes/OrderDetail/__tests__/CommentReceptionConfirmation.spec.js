import React from 'react';

// NOTE: the named export isn't a connected component
import { CommentReceptionConfirmation } from '../CommentReceptionConfirmation';

it('renders CommentReceptionConfirmation correctly', () => {
  expect(shallow(<CommentReceptionConfirmation />)).toMatchSnapshot();
});
