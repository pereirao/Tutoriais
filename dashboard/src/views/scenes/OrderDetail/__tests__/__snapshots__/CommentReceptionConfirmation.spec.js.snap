// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CommentReceptionConfirmation correctly 1`] = `
<div
  className="commentReceptionConfirmation"
>
  <div
    className="receptionConfirmationButton"
    onClick={[Function]}
  >
    I NEED HELP, PLEASE CALL
  </div>
  <div
    className="receptionConfirmationButton"
    onClick={[Function]}
  >
    OK, I UNDERSTAND
  </div>
</div>
`;
